container_registry/container_analysis/src/sample/samples.go

```diff
@@ -0,0 +1,247 @@
+
+// [START get_occurrence]
+
+// getOccurrence retrieves an Occurrence from the server.
+func getOccurrence(ctx context.Context, client *containeranalysis.GrafeasV1Beta1Client, occurrenceName string) (*grafeaspb.Occurrence, error) {
+	req := &grafeaspb.GetOccurrenceRequest{Name: occurrenceName}
+	return client.GetOccurrence(ctx, req)
+}
+
+// [END get_occurrence]
+
+// [START discovery_info]
+
+// getDiscoveryInfo retrieves and prints the Discovery Occurrence created for a specified image.
+// The Discovery Occurrence contains information about the initial scan on the image.
+func getDiscoveryInfo(ctx context.Context, client *containeranalysis.GrafeasV1Beta1Client, imageUrl, projectID string) error {
+	filterStr := `kind="DISCOVERY" AND resourceUrl="` + imageUrl + `"`
+	projectName := "projects/" + projectID
+
+	req := &grafeaspb.ListOccurrencesRequest{Parent: projectName, Filter: filterStr}
+	it := client.ListOccurrences(ctx, req)
+	for {
+		occ, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		fmt.Println(occ)
+	}
+	return nil
+}
+
+// [END discovery_info]
+
+// [START occurrences_for_note]
+
+// getOccurrencesForNote retrieves all the Occurrences associated with a specified Note.
+// Here, all Occurrences are printed and counted.
+func getOccurrencesForNote(ctx context.Context, client *containeranalysis.GrafeasV1Beta1Client, noteID, projectID string) (int, error) {
+	noteName := "projects/" + projectID + "/notes/" + noteID
+
+	req := &grafeaspb.ListNoteOccurrencesRequest{Name: noteName}
+	it := client.ListNoteOccurrences(ctx, req)
+	count := 0
+	for {
+		_, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return -1, err
+		}
+		// Write custom code to process each Occurrence here.
+		count = count + 1
+	}
+	return count, nil
+}
+
+// [END occurrences_for_note]
+
+// [START occurrences_for_image]
+
+// getOccurrencesForImage retrieves all the Occurrences associated with a specified image.
+// Here, all Occurrences are simply printed and counted.
+func getOccurrencesForImage(ctx context.Context, client *containeranalysis.GrafeasV1Beta1Client, imageUrl, projectID string) (int, error) {
+	filterStr := `resourceUrl="` + imageUrl + `"`
+	project := "projects/" + projectID
+
+	req := &grafeaspb.ListOccurrencesRequest{Parent: project, Filter: filterStr}
+	it := client.ListOccurrences(ctx, req)
+	count := 0
+	for {
+		_, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return -1, err
+		}
+		// Write custom code to process each Occurrence here.
+		count = count + 1
+	}
+	return count, nil
+}
+
+// [END occurrences_for_image]
+
+// [START pubsub]
+
+// occurrencePubsub handles incoming Occurrences using a Cloud Pub/Sub subscription.
+func occurrencePubsub(ctx context.Context, subscriptionID string, timeout int, projectID string) (int, error) {
+	toctx, _ := context.WithTimeout(ctx, time.Duration(timeout)*time.Second)
```

Comment:
Re-assign the context to the "ctx" variable. No need to create a new variable, since the original ctx isn't used.